natlas-server/app/models/scope_item.py

```diff
@@ -95,40 +95,82 @@ def parse_import_line(line):
         ip = ScopeItem.validate_ip(ip)
         return ip, tags
 
+    @staticmethod
+    def extract_import_tags(import_list: list) -> set:
+        tags = set()
+        for line in import_list:
+            split = line.split(",")
+            if len(split) > 1:
```

Comment:
If there's no commas, there's no error, there's just a line that doesn't have any tags. If there's a comma but nothing after it, it'll be an empty string, of which there can only be one. Anything else is a valid tag, it's just a name and there are currently no restrictions on the name (other than character length, which we aren't validating right now). This is just preloading the tags into the database so that creating the associations afterwards is faster (we're only trying to create each tag a single time, instead of for every line that the tag appears on).

Reply:
I'm not usually a fan of silently swallowed invalid data since people start to give garbage data. What do you think about throwing an error of the line isn't formatted right?

Reply:
I don't think there's a clear "error" situation right now except when there's an empty tag. No commas is not an error, tags are just used to enhance information about the scope being imported and there isn't a tight relationship that says the scope needs to be tagged. I'd like to move to a json or yaml import eventually to handle imports as we begin to augment import data more, but for now I think we should leave this alone.